FinancialTransactionsApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -118,5 +130,88 @@ public async Task UpdateAsync(Transaction transaction)
+        {
+
+            var firstDay = new DateTime(DateTime.Now.Year, 1, 1);
+            var lastDay = new DateTime(DateTime.Now.Year, 12, 31);
+            string startDate = firstDay.ToString(AWSSDKUtils.ISO8601DateFormat);
+
+            string endDate = lastDay.ToString(AWSSDKUtils.ISO8601DateFormat);
+            QueryRequest queryRequest = new QueryRequest
+            {
+                TableName = "Transactions",
+                KeyConditionExpression = "pk = :V_pk",
+                FilterExpression = "target_id =:V_target_id AND transaction_date BETWEEN :V_startDate AND :V_endDate",
+                ExpressionAttributeValues = new Dictionary<string, AttributeValue>
+                    {
+                     {":V_pk", new AttributeValue {S = Pk}},
+                     {":V_target_id", new AttributeValue {S = query.TargetId.ToString()}},
+                     {":V_startDate", new AttributeValue { S = startDate }},
+                     {":V_endDate", new AttributeValue { S = endDate }}
+                    }
+            };
+
+            if (query.TransactionType != null)
+            {
+                queryRequest.FilterExpression += " AND transaction_type = :V_transaction_type";
+                queryRequest.ExpressionAttributeValues.Add(":V_transaction_type", new AttributeValue { S = query.TransactionType.ToString() });
+            }
+
+            var result = await _amazonDynamoDb.QueryAsync(queryRequest).ConfigureAwait(false);
+            var transactions = result.ToTransactions();
+            return transactions;
+        }
+
+        public async Task<List<Transaction>> GetAllTransactionRecordAsync(ExportTransactionQuery query)
+        {
+            DateTime firstDay;
+            DateTime lastDay;
+            if (query.StatementType == StatementType.Quaterly)
+            {
+
+                firstDay = DateTime.UtcNow;
+                lastDay = firstDay.AddMonths(-3).AddDays(-1);
+
+            }
+            else
+            {
+                firstDay = DateTime.UtcNow;
+                lastDay = firstDay.AddMonths(-12);
+            }
+
+            var config = new DynamoDBOperationConfig()
+            {
+                QueryFilter = new List<ScanCondition>() {
+                    new ScanCondition("TargetId", ScanOperator.Equal, query.TargetId),
+                    new ScanCondition("TransactionDate", ScanOperator.Between,lastDay,firstDay)
+                }
+            };
+            var response = await _dynamoDbContext.QueryAsync<TransactionDbEntity>(Pk, config).GetRemainingAsync().ConfigureAwait(false);
+            if (response.Count < 1) return new List<Transaction>();
+            var result = response.Select(x => x.ToDomain()).OrderBy(_ => _.TransactionDate).ToList();
+            return result;
+        }
+
+        public async Task<List<Transaction>> GetAllTransactionByDateAsync(TransactionExportRequest request)
```

Comment:
How does that method differ from the above?

Reply:
Code refactored